src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -230,22 +231,45 @@ protected Task Stop()
                     }
                 });
         }
-
-        protected async Task HandleShowOnlineHelpRequest(
+        protected async Task HandleShowHelpRequest(
             string helpParams,
             RequestContext<object> requestContext)
         {
-            if (helpParams == null) { helpParams = "get-help"; }
-
-            var psCommand = new PSCommand();
-            psCommand.AddCommand("Get-Help");
-            psCommand.AddArgument(helpParams);
-            psCommand.AddParameter("Online");
-
-            await editorSession.PowerShellContext.ExecuteCommand<object>(psCommand);
-
+            const string CheckHelpScript = @"
+                [CmdletBinding()]
+                param (
+                    [String]$CommandName
+                )
+                try {
+                    $null = Microsoft.PowerShell.Core\Get-Command $CommandName -ErrorAction Stop
+                } catch [System.Management.Automation.CommandNotFoundException] {
+                    $PSCmdlet.ThrowTerminatingError($PSItem)
+                }
+                try {
+                    $null = Microsoft.PowerShell.Core\Get-Help $CommandName -Online
+                } catch [System.Management.Automation.PSInvalidOperationException] {
+                    Microsoft.PowerShell.Core\Get-Help $CommandName -Full
+                }";
+            if (string.IsNullOrEmpty(helpParams)) { helpParams = "Get-Help"; }
+
+            PSCommand checkHelpPSCommand = new PSCommand()
+                .AddScript(CheckHelpScript, useLocalScope: true)
+                .AddArgument(helpParams);
+            await editorSession.PowerShellContext.ExecuteCommand<PSObject>(checkHelpPSCommand, sendOutputToHost: true);
             await requestContext.SendResult(null);
         }
+        protected async Task HandleShowOnlineHelpRequest(
+            string helpParams,
+            RequestContext<object> requestContext
+        )
+        {
+            const string deprecatedWarning = @"
+            Write-Warning ""'powerShell/showOnlineHelp' has been deprecated. Use 'powerShell/showHelp' instead.""";
```

Comment:
Instead of `AddScript` you could use `AddCommand` here instead.

```csharp
pwsh.AddCommand("Microsoft.PowerShell.Utility\\Write-Verbose")
    .AddParameter("Message", "'powerShell/showOnlineHelp' has been deprecated. Use 'powerShell/showHelp instead.");
```

Using command instead allows it to create a command processor for that command specifically, instead of having to parse a script.